### PR TITLE
Merged, thanks!

### PR DESCRIPTION
**Pull request by electrum**

### CODE REVIEW

File: presto-hive/src/main/java/io/prestosql/plugin/hive/HiveWriterFactory.java [hunk 1]
```diff
@@ -589,7 +588,7 @@ private void validateSchema(Optional<String> partitionName, Properties schema)
                         schemaName,
                         tableName,
                         inputHiveType,
-                        partitionName,
+                        partitionName.orElse(""),
```
Comment (findepi) [inline presto-hive/src/main/java/io/prestosql/plugin/hive/HiveWriterFactory.java]: i am not sure i like `.. but partition '' declared column` in the final err msg.

but, the `partitionName` must exist at this point, so `partitionName.orElseThrow` would be appropriate

File: presto-hive/src/main/java/io/prestosql/plugin/hive/HiveWriterFactory.java [hunk 2]
```diff
@@ -561,12 +560,12 @@ private void validateSchema(Optional<String> partitionName, Properties schema)
...
         Set<String> missingColumns = Sets.difference(inputColumnMap.keySet(), new HashSet<>(fileColumnNames));
         if (!missingColumns.isEmpty()) {
-            throw new PrestoException(NOT_FOUND, format("Table '%s.%s' does not have columns '%s'", schema, tableName, missingColumns));
+            throw new PrestoException(HIVE_INVALID_METADATA, format("Table '%s.%s' does not have columns %s", schemaName, tableName, missingColumns));
         }
         if (fileColumnNames.size() != fileColumnHiveTypes.size()) {
             throw new PrestoException(HIVE_INVALID_METADATA, format(
                     "Partition '%s' in table '%s.%s' has mismatched metadata for column names and types",
-                    partitionName,
+                    partitionName.orElse(""),
```
Comment (findepi) [inline presto-hive/src/main/java/io/prestosql/plugin/hive/HiveWriterFactory.java]: same here _probably_

File: presto-hive/src/main/java/io/prestosql/plugin/hive/HiveWriterFactory.java (hunk 1 above)
Comment (electrum) [reply presto-hive/src/main/java/io/prestosql/plugin/hive/HiveWriterFactory.java]: I think that’s true, but I’m not comfortable with restructuring the code so the optional can be removed (especially without tests).

Comment (electrum) [reply presto-hive/src/main/java/io/prestosql/plugin/hive/HiveWriterFactory.java]: I think that’s true, but I’m not comfortable with restructuring the code so the optional can be removed (especially without tests).

**Comment by electrum** [comment]
The purpose of this PR is to

1. Use the correct error code for the exception.
2. Remove the `Optional[]` wrapper from the string.

It does not address the issue that the error message is wrong for unpartitioned tables, assuming that case can even exist, but that's outside the scope of this PR and shouldn't block fixing the above.

File: presto-hive/src/main/java/io/prestosql/plugin/hive/HiveWriterFactory.java (hunk 1 above)
Comment (findepi) [reply presto-hive/src/main/java/io/prestosql/plugin/hive/HiveWriterFactory.java]: Can you add a code comment maybe like `// partitionName is expected to exist`` ?

**Pull request closed without merging**